sunfish/lib/core.py

```
@@ -31,17 +28,81 @@ def __init__(self, conf):
+        # Plugins are implemented as namespaced packages and must be placed in a folder at the top of the project named
+        # "sunfish_plugins", with subfolders named "storage" and/or "events_handlers" and/or "objects_handlers".
+        # The python packages defined inside each subfolder are totally user defined.
+        # ── sunfish_plugins
+        #    ├── storage
+        #    │   └──my_storage_package     <--- User defined
+        #    │      ├── __init__.py
+        #    │      └── my_storage_backend.py
+        #    └── events_handlers
+        #    │   └──my_handler_package     <--- User defined
+        #    │      ├── __init__.py
+        #    │      └── my_handler.py
+        #    ├── object_handlers
+        #    │   └──my_objects_handler_package     <--- User defined
+        #    │      ├── __init__.py
+        #    │      └── my_objects_handler.py
+
+        # When initializing the Sunfish libraries can load their storage or event handler plugin by specifying them in
+        # the configuration as in the below example:
+        #
+        # "storage_backend" : {
+        #         "module_name": "storage.my_storage_package.my_storage_backend",
+        #         "class_name": "StorageBackend"
+        # },
+        # "event_handler" : {
+        #         "module_name": "events_handlers.my_handler_package.my_handler",
+        #         "class_name": "StorageBackend"
+        # },
+        # "objects_handler" : {
+        #         "module_name": "objects_handlers.my_objects_handler_package.my_objects_handler",
+        #         "class_name": "StorageBackend"
+        # },
+        #
+        # In all cases "class_name" represents the name of the class that is initialized and implements the respective
+        # interface.
+
+        # Default storage plugin loaded if nothing is specified in the configuration
+        if "storage_backend" not in conf:
```

I would change the flow of all three of these loaders.  Instead of conf or default, try load, fail if not found, I would do test conf, load if in conf, check to see if found, default to fallback if no conf or not found.  That way a broken conf file won't keep us from loading the default and being (hopefully) minimally useful

---

Good point. I will make it so that it starts from conf, if available, and fall-back to default before failing. Just FYI, the text in the comment is showing the conf file and not the actual load for loading plugins.

---

@dledford done